app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardFinanceWidget.kt

```diff
@@ -212,3 +213,39 @@ fun ExpenseItem(expense: Expense, currencyCode: String) {
             modifier = Modifier.padding(8.dp).testTag("expenseAmount" + expense.expenseId))
       }
 }
+
+/**
+ * Function to format the total expense amount to a more readable format. The function takes the
+ * total expense amount as a Double and returns a formatted String. The function formats the amount
+ * to a maximum of 2 decimal places and adds a suffix of M, B, or T based on the number of digits in
+ * the amount.
+ *
+ * @param totalexpense The total expense amount as a Double.
+ */
+private fun formatExpense(totalexpense: Double): String {
```

Comment:
Consider moving this function out of the screen and into the ViewModel for better encapsulation and easier testing with unit tests.